### PR TITLE
feat: check all trips are used in stop_times.txt

### LINKED ISSUE #519: Trips must be used in `stop_times.txt`
**Describe the solution you'd like**
Validate all trips defined are used in `stop_times.txt`

**Additional context**
`v1.4.0` implementation:
- https://github.com/MobilityData/gtfs-validator/blob/v1.4.0/usecase/src/main/java/org/mobilitydata/gtfsvalidator/usecase/ValidateTripUsage.java
- tests: https://github.com/MobilityData/gtfs-validator/blob/v1.4.0/usecase/src/test/java/org/mobilitydata/gtfsvalidator/usecase/ValidateTripUsageTest.java

### PR DESCRIPTION
**Pull request by lionel-nj**
closes #519

**Summary:**

This PR implements a new validation rule: all trips must be used in stop_times.txt

**Expected behavior:** 

- trips that are not used in `stop_times.txt` should generate a notice
- trips that are used in `stop_times.txt` should not generate a notice

New notice: `UnusedTripNotice`

Please make sure these boxes are checked before submitting your pull request - thanks!

- [x] Run the unit tests with `gradle test` to make sure you didn't break anything
- [x] Format the title like "feat: -new feature short description-" (PR title must follow the [conventional commit specification](https://www.conventionalcommits.org/en/v1.0.0/))
- [x] Linked all relevant issues
- ~[ ] Include screenshot(s) showing how this pull request works and fixes the issue(s)~

### CODE REVIEW

File: main/src/main/java/org/mobilitydata/gtfsvalidator/notice/UnusedTripNotice.java [hunk 1]
```diff
@@ -0,0 +1,33 @@
+/*
+ * Copyright 2020 Google LLC
```
Comment (barbeau) [inline main/src/main/java/org/mobilitydata/gtfsvalidator/notice/UnusedTripNotice.java:2]: Copyright should be MobilityData if Google didn't write the code.

File: main/src/main/java/org/mobilitydata/gtfsvalidator/validator/TripUsageValidator.java [hunk 2]
```diff
@@ -0,0 +1,52 @@
...
+ */
+@GtfsValidator
+public class TripUsageValidator extends FileValidator {
+  @Inject GtfsTripTableContainer tripTable;
+
+  @Inject GtfsStopTimeTableContainer stopTimeTable;
+
+  @Override
+  public void validate(NoticeContainer noticeContainer) {
+    // Do not report the same shape_id multiple times.
```
Comment (barbeau) [inline main/src/main/java/org/mobilitydata/gtfsvalidator/validator/TripUsageValidator.java:43]:
```suggestion
    // Do not report the same trip_id multiple times.
```

**Review by barbeau — COMMENTED** [review]
@lionel-nj This looks good! Two minor comments.

File: main/src/main/java/org/mobilitydata/gtfsvalidator/notice/UnusedTripNotice.java (hunk 1 above)
Comment (lionel-nj) [reply main/src/main/java/org/mobilitydata/gtfsvalidator/notice/UnusedTripNotice.java:2]: Indeed, thanks.

**Review by barbeau — APPROVED** [review]
LGTM!

**Pull request merged**